### PR TITLE
filters for parsing i18n yml to js for Ember.String.loc or i18n-js

### PR DESCRIPTION
**Pull request by randym**
Came across a need for this in travis-ci. If you think it would be useful for the rest of the community please feel free to merge.

It takes yml files and parses them out into js using syntax appropriate for Ember.STRINGS (for using loc) or i18n.translations (for using i18n-js) based on a use_i18n_js option.

### CODE REVIEW

**Comment by ahawkins** [comment]
+1. I did this in Iridium. I didn't even know this PR requested.

File: lib/rake-pipeline-web-filters/ember_i18n_filter.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# encoding: UTF-8
```
Comment (wycats) [inline lib/rake-pipeline-web-filters/ember_i18n_filter.rb]: You should only need to specify the encoding in a file that actually contains UTF-8 characters. Why did you need to include it here?

Comment (randym) [inline lib/rake-pipeline-web-filters/ember_i18n_filter.rb]: Please accept my apologies. It is a bad habit developed from living and working in Japan for so many years. As it only needs to be in the spec file, I will remove this immediately. Out of curiosity, and at the risk of showing my own ignorance, are there potential issues for other developers when the encoding is unnecessarily specified?

**Comment by ppcano** [comment]
I am in favor of having a i18n convention on EmberJS, otherwise each APP will implement it on different ways. I have been using a similar approach with rake-pipeline and i18n.

**Comment by wagenet** [comment]
@twinturbo How does this compare to what you did? Might be nice to have something in RPWF.

**Comment by ahawkins** [comment]
@wagenet see source here: https://github.com/radiumsoftware/iridium/blob/master/lib/rake-pipeline-web-filters/i18n_filter.rb

``` ruby
match "locales/**/*.ym" do
  i18n "locales.js"
end
```

Very simple filter and straight to the point. The `destination` could be configurable (something other than `I18n.translations`).

**Comment by randym** [comment]
While I have no basis to comment on the Worth noting that Ember.STRING and Ember.string.loc output was a bit of a PITA. I' be very happy to see EmberJS incorporate i18n-js from fnando as it would greatly simplify this filter.

**Comment by randym** [comment]
moving this into a separate gem.

**Comment by ahawkins** [comment]
Hmm. This can be in the gem. I think some sort of functionality like this
is useful. I'll just try out irdiums implementation in a pr

**Comment by randym** [comment]
rake-pipeline-i18n-filters released last week.
https://rubygems.org/gems/rake-pipeline-i18n-filters
